### PR TITLE
Modify maintenance resp

### PR DESCRIPTION
**Pull request by yufenghe22**
In this PR, I considered the `mrc_leaf` and `mrc_stem` to be equal and re-did the optimization for the 2002-2006 datasets.
The updated results for comparing with the observed biomasses can be found [here](https://github.com/Matthews-Research-Group/soil-tipping-bucket/blob/main/run/optimization/figs/fig_new_parameters_v2.pdf).

The reasons are as follows,

- Treating them separately would cause over-fitting on the LD11's data. Making `mrc_leaf` and `mrc_stem` equal solved this problem.
- In the literatures, discussions are often made on the shoot respiration including both leaf and stem. This is also true in some other [models](https://doi.org/10.1016/j.fcr.2010.07.007).
- I also tested a further simplification by use only one parameter for  `mrc_leaf`, `mrc_stem` and `mrc_root`. This however did not work well. Having even a small `mrc` on the leaf and stem can cause root to continue decreasing when optimizing for the LD11’s datasets, which does NOT agree with the observed root data that has no clear decreasing trends. In our model, root stops getting partitioned carbon after the initial short period. My guess is that the logistic function is not suitable to describe carbon partition to root (`kroot`). `kroot` might be max initially, and suddenly drop to a steady value after that initial period, which might be better described by a discontinuous function. Nevertheless, this is not something to address in this PR.

@leighmatth

### CODE REVIEW

File: data/soybean.R [hunk 1]
```diff
@@ -168,30 +168,30 @@ soybean <- list(
...
-        grc_root                        = 0.0025,    # dimensionless, optimized 
+        grc_stem                        = 0.02828,    # dimensionless, optimized. growth respiration coefficient 
+        grc_root                        = 0.00270,    # dimensionless, optimized 
         
         # maintenance respiration module
-        mrc_leaf                        = 3.7e-4,    # hour^-1, maintenance respiration coefficient. optimized 
-        mrc_stem                        = 4.6e-4,    # hour^-1, optimized 
-        mrc_root                        = 1e-5,      # hour^-1, optimized 
-        mrc_grain                       = 1e-5,      # hour^-1, optimized 
+        mrc_leaf                        = 0.00048836,    # hour^-1, maintenance respiration coefficient. optimized
```
Comment (leavesandlemmas) [inline data/soybean.R:175]: The units here are not written the same as at line 90 in src/module_library/maintenance_respiration.h  These have the same units, right? So `kg / kg / hr` would be consistent with how units are written elsewhere

**Review by leavesandlemmas — APPROVED** [review]
Just a few comments on how units are written. 

I think the carbon allocation is going to change at some point so that couldl fixed the issue with the roots not getting enough new carbon to keep the same mass.

**Review by eloch216 — APPROVED** [review]
I think it makes sense to use the same value for `mrc_leaf` and `mrc_stem`, and this seems like an improvement to the model. I also agree with Scott's comments regarding the units.

File: src/module_library/maintenance_respiration.h [hunk 2]
```diff
@@ -87,10 +87,10 @@ string_vector maintenance_respiration::get_inputs()
         "Shell",      // Mg / ha
         "Grain",      // Mg / ha
         "temp",       // degree C
-        "mrc_leaf",   // degree C
-        "mrc_stem",   // degree C
-        "mrc_root",   // degree C
-        "mrc_grain"   // degree C
+        "mrc_leaf",   // kg kg-1 h-1
```
[the remark this answers is not in the document]
Comment (yufenghe22) [reply src/module_library/maintenance_respiration.h]: Done!

**Comment by yufenghe22** [comment]
Will merge if no further comments by the end of today.

**Comment by eloch216** [comment]
> Will merge if no further comments by the end of today.

I did think of one other thing -- it would be good to add a new `item` in the `details` section of `man/soybean.Rd` giving a brief overview of the recent soybean model reparameterization, describing changes from this PR and PR #120.

**Comment by yufenghe22** [comment]
> I did think of one other thing -- it would be good to add a new `item` in the `details` section of `man/soybean.Rd` giving a brief overview of the recent soybean model reparameterization, describing changes from this PR and PR #120.

Added! Hope it reads clear.

**Pull request merged**